src/nl/hannahsten/texifyidea/file/SaveImageFromWebDialog.kt

```diff
@@ -0,0 +1,300 @@
+package nl.hannahsten.texifyidea.file
+
+import com.intellij.openapi.fileChooser.FileChooserDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.ui.*
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.ui.components.JBLabel
+import com.intellij.ui.components.JBTextField
+import com.intellij.util.ui.ImageUtil
+import nl.hannahsten.texifyidea.ui.ImagePanel
+import nl.hannahsten.texifyidea.util.addLabeledComponent
+import nl.hannahsten.texifyidea.util.formatAsFileName
+import java.awt.BorderLayout
+import java.awt.Dimension
+import java.awt.Image
+import java.awt.datatransfer.DataFlavor
+import java.awt.image.BufferedImage
+import java.io.File
+import java.util.*
+import javax.imageio.ImageIO
+import javax.swing.BoxLayout
+import javax.swing.JPanel
+import javax.swing.border.EmptyBorder
+
+/**
+ * @author Hannah Schellekens
+ */
+class SaveImageFromWebDialog(
```

Comment:
@jojo2357 what? You copy pasted a whole 300 line class and just removed a couple of lines you didn't need?

Reply:
You may have noticed, but this does not have a `Transferable` like the other class uses. Maybe we can re-merge them using abstraction at some point?